ksqldb-functional-tests/src/test/resources/query-validation-tests/joins.json

```diff
@@ -2298,6 +2342,159 @@
         "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "Incompatible key formats. `T1` has KAFKA while `T2` has DELIMITED.\nCorrect the key format by creating a copy of the table with the correct key format. For example:\n\tCREATE TABLE T_COPY\n\t WITH (KEY_FORMAT = <required format>, <other key format config>)\n\t AS SELECT * FROM T;"
       }
+    },
+    {
+      "name": "stream-steam key-to-key - SR-enabled key format",
```

Comment:
> we might want to introduce a pre-topics node here so that we have control over the actual schema in schema registry. I think we also have the ability to ensure the schema at the end.

Yes, I've been thinking this as well. Wanted to open the PR first in order to sanity-check the approach -- since we're on the same page I'll pursue beefing up the tests 👍 

> I think it also makes sense to add tests for:
> 
> 1. schemas are identical, should repartition anyway
> 2. schemas are logically identical but physically different
> 3. schemas are logically different, should fail (in table-table joins anyway)

The new tests I added are already case (1). Tests for case (3) exist for non-SR-enabled formats, but I'll add one for an SR-enabled format. I'll add (2) and also tests with SR-enabled value formats as a sanity check.

> 🚂

Haha! I fell victim to copy-paste from existing tests. I made the update, which required renaming existing historic plans so this PR diff just became a lot larger in terms of number of lines.

Reply:
Added the different tests discussed above. Good call on testing table-table join with nulls -- turns out we've got a bug. The newly added test won't pass until https://github.com/confluentinc/ksql/pull/6647 is merged.